### PR TITLE
Wallet SDK UI/UX Guide

### PR DESCRIPTION
**Pull request by lzanchin-immutable**
The guide with the recommended UI/UX expected when partners use the Wallet SDK.

## Description

The guidelines we will recommend to partners who adopt the Wallet-SDK-web in conjunction with the Core-SDK.

### CODE REVIEW

File: docs/main/guides/wallet-sdk-ui-guide.md [hunk 1]
```diff
@@ -0,0 +1,329 @@
...
+                <ul>
+                    <li>Surface purchase price</li>
+                    <li>Surface purchase currency (assuming you support multiple currencies)</li>
+                    <li>Maker / taker fee breakdown</li>
+                </ul>
+                Useful to have:
+                <ul>
+                    <li>User feedback that purchase was successful</li>
+                    <li>Send users to add funds flow if they have insufficient funds</li>
+                </ul>
```
Comment (sprngbk) [inline docs/main/guides/wallet-sdk-ui-guide.md:145]:
```suggestion
                <ul>
                    <li>Display the purchase price</li>
                    <li>Display the purchase currency</li>
                    <li>Provide breakdown of any maker and taker fees that apply</li>
                </ul>
                Not mandatory, but useful to have:
                <ul>
                    <li>User feedback that purchase was successful</li>
                    <li>Send users to an "add funds" flow if they have insufficient funds</li>
                </ul>    
```

File: docs/main/guides/wallet-sdk-ui-guide.md [hunk 2]
```diff
@@ -0,0 +1,329 @@
...
+                    <li>Send users to add funds flow if they have insufficient funds</li>
+                </ul>    
+            </td>
+        </tr>
+</table>
+
+Requirement for custom UI:
+
+- View asset listing amount
+- Payment flow entry points (Deposit from L1, Onramp with Moonpay, Pay with CC through Moonpay).
```
Comment (sprngbk) [inline docs/main/guides/wallet-sdk-ui-guide.md:153]: Where are these points meant to sit?

File: docs/main/guides/wallet-sdk-ui-guide.md [hunk 3]
```diff
@@ -0,0 +1,329 @@
...
+        <tr>
+            <td>Add funds flow</td>
+            <td>
+                <img src="/img/wallet-sdk-ui-guide/add-funds.png" alt="Add Funds" width="50%" height="50%" />
+            </td>
+            <td>
+                <strong>Partner must</strong>
+                <ul>
+                    <li>Surface add funds flow</li>
+                </ul>
```
Comment (sprngbk) [inline docs/main/guides/wallet-sdk-ui-guide.md:172]:
```suggestion
                <ul>
                    <li>Provide an "add funds" flow</li>
                </ul>
```

File: docs/main/guides/wallet-sdk-ui-guide.md [hunk 4]
```diff
@@ -0,0 +1,329 @@
...
+                <ul>
+                    <li>Collect currency type (assuming multiple currencies available)</li>
+                    <li>Collect amount to be deposited</li>
+                </ul>
+                Useful to have:
+                <ul>
+                    <li>Icon for different currency types</li>
+                    <li>Available L1 balance to be deposited</li>
+                    <li> User feedback that deposit was successful</li>
+                </ul>
```
Comment (sprngbk) [inline docs/main/guides/wallet-sdk-ui-guide.md:191]:
```suggestion
                <ul>
                    <li>Provide a field for the user to input the currency type</li>
                    <li>Provide a field for the user to input the amount they want to deposit</li>
                </ul>
                Not mandatory, but useful to have:
                <ul>
                    <li>Icons representing different currency types</li>
                    <li>Available L1 balance to be deposited</li>
                    <li> Feedback when a deposit is successful</li>
                </ul>    
```

File: docs/main/guides/wallet-sdk-ui-guide.md [hunk 5]
```diff
@@ -0,0 +1,329 @@
...
+                <ul>
+                    <li>Icon for different currency types</li>
+                    <li>Available L1 balance to be deposited</li>
+                    <li> User feedback that deposit was successful</li>
+                </ul>    
+            </td>
+        </tr>
+</table>
+
+# L2 to L1 Withdraw
```
Comment (sprngbk) [inline docs/main/guides/wallet-sdk-ui-guide.md:196]:
```suggestion
# L2 to L1 withdrawal
```

File: docs/main/guides/wallet-sdk-ui-guide.md [hunk 6]
```diff
@@ -0,0 +1,329 @@
...
+                <li>Icon for different currency types</li>
+                <li>Available L2 balance to be deposited</li>
+                <li>Notice that withdrawal will require L1 gas to be paid</li>
+                <li>Notice that withdrawal has 2 stage: preparation & execution of withdrawal (more information available <a href="https://support.immutable.com/hc/en-us/articles/1500003550482">here</a>)</li>
+            </ul>
+        </td>
+    </tr>
+</table>
+
+# Fiat On-Ramp [ workflow to be added ]
```
Comment (sprngbk) [inline docs/main/guides/wallet-sdk-ui-guide.md:228]: Does this mean that the workflow does not yet exist in the Wallet SDK?

File: docs/main/guides/wallet-sdk-ui-guide.md [hunk 7]
```diff
@@ -0,0 +1,329 @@
...
+        <td>Fiat onramp</td>
+        <td>
+            <img src="/img/wallet-sdk-ui-guide/fiat-onramp.png" alt="Add Funds" width="50%" height="50%" />
+        </td>
+        <td>
+            <strong>Partner must</strong>
+            <ul>
+                <li>Trigger the onramp flow</li>
+            </ul>
+            Moonpay widget will take care of the rest
```
Comment (sprngbk) [inline docs/main/guides/wallet-sdk-ui-guide.md:246]:
```suggestion
            <ul>
                <li>Trigger the onramp flow</li>
            </ul>
            The Moonpay widget will take care of the rest.
```

File: docs/main/guides/wallet-sdk-ui-guide.md [hunk 8]
```diff
@@ -0,0 +1,329 @@
...
+        <td>
+            <strong>Partner must</strong>
+            <ul>
+                <li>Trigger the offramp flow</li>
+            </ul>
+            Moonpay widget will take care of the rest.<br/>
+            Useful to have:
+            <ul>
+                <li>Moonpay Offramp is only available to users in the US, UK, and EU</li>
+            </ul>
```
Comment (sprngbk) [inline docs/main/guides/wallet-sdk-ui-guide.md:273]:
```suggestion
            <ul>
                <li>Trigger the offramp flow</li>
            </ul>
            The Moonpay widget will take care of the rest.

            Not mandatory, but useful to have:
            <ul>
                <li>A note informing the user that the Moonpay offramp is only available to users in the US, UK, and EU.</li>
            </ul>
```

File: docs/main/guides/wallet-sdk-ui-guide.md [hunk 9]
```diff
@@ -0,0 +1,329 @@
...
+        <td>
+            <strong>Partner must</strong>
+            <ul>
+                <li>Provide user feedback about specific errors</li>
+                <li> Error list can be found <a href="https://docs.x.immutable.com/docs/link-errors/">here</a></li>
+            </ul>
+            Useful to have:
+            <ul>
+                <li>Link to Immutable support page</li>
+            </ul>
```
Comment (sprngbk) [inline docs/main/guides/wallet-sdk-ui-guide.md:300]:
```suggestion
            <ul>
                <li>Provide user feedback about specific errors (full list of errors can be found <a href="https://docs.x.immutable.com/docs/link-errors/">here</a>)</li></li>
            </ul>
            Not mandatory, but useful to have:
            <ul>
                <li>Link to Immutable support page</li>
            </ul>
```

File: docs/main/guides/wallet-sdk-ui-guide.md [hunk 10]
```diff
@@ -0,0 +1,329 @@
...
+        </td>
+        <td>
+            <strong>Partner must</strong>
+            <ul>
+                <li>Provide entry point for transactions</li>
+                <li>Transaction ID</li>
+                <li>Asset</li>
+                <li>Action</li>
+                <li>Cost</li>
+            </ul>
```
Comment (sprngbk) [inline docs/main/guides/wallet-sdk-ui-guide.md:326]:
```suggestion
            <ul>
                <li>Provide users with the ability to view their transaction history</li>
                <li>For each transaction, they must provide information about the transaction ID, asset details, type of transaction and the costs involved.</li>
            </ul>
```

**Review by sprngbk — CHANGES_REQUESTED** [review]
Thanks Leandro, just left some comments and suggestions.

Just a couple of things to note:
* I removed references to "partners" and replaced with "application" where necessary because I think it's a more generic term and this guide will also be used by our self-serve users
* I removed "Partner must have" phrases because that is implied as it is in the "Requirements" column. I think it's clear that these are mandatory when we highlight the practices that are "not mandatory, but useful to include..."
* Can we please use **sentence casing** throughout this guide according to our style manual [here](https://immutable.atlassian.net/wiki/spaces/PPS/pages/1937278619/Style+manual#id-%E2%9C%92%EF%B8%8F%F0%9F%93%84Stylemanual-Capitalizethefirstwordinasentenceandinheadings)?

File: docs/main/guides/wallet-sdk-ui-guide.md [hunk 11]
```diff
@@ -0,0 +1,329 @@
...
+            <td>
+                Useful to include:
+                <ul>
+                    <li>Explanation of why there are multiple signature requests from the non-custodial wallet</li>
+                    <ul>
+                        <li>Connecting an account</li>
+                        <li>Signature request for access</li>
+                        <li>Setting up a IMX Key (needed on the first time only)</li>
+                    </ul>
+                    <li>If having both Metamask and Gamestop as wallet options, explain somehow that having both plugins at the same time can cause issues to the wallet connection</li>
```
[the remark this answers is not in the document]
Comment (lzanchin-immutable) [reply docs/main/guides/wallet-sdk-ui-guide.md:55]: I will include a link to the support page which have more details. (https://support.immutable.com/hc/en-us/articles/5160531224079-Managing-multiple-wallet-extensions-for-Immutable)

File: docs/main/guides/wallet-sdk-ui-guide.md (hunk 6 above)
Comment (lzanchin-immutable) [reply docs/main/guides/wallet-sdk-ui-guide.md:228]: Correct

**Review by sprngbk — APPROVED** [review]
Looks great, thanks Leandro!

**Pull request merged**